### ISSUE TITLE
Recent commit introduces a build failure

### ISSUE THREAD

**Issue by Pointedstick** [endless-sky/endless-sky#1581]
Latest code fails to build. Font.cpp, line 239:

```
if(prevWorks != nextWorks && abs(nextChars - prevChars) == 1)     ❌Call to 'abs' is ambiguous
```

This was introduced in be5f7c47fc8d39ee9a40a0a4a07217f5bfbd9b4d.

Works if you change it to `fabs()`.

**Comment by Pointedstick** [comment]
Addressed in pull request https://github.com/endless-sky/endless-sky/pull/1582

**Comment by Bladewood** [comment]
Odd. It builds just fine on my end.

**Comment by MarcelineVQ** [comment]
The issue is that in `abs(nextChars - prevChars)` nextChars and prevChars are unsigned which means abs is redundant and possibly an error can creep in from numbers overflowing on a subtraction.

The side effect of this is that it's not sure what function to use because size_t is unsigned which abs doesn't have an overload for, for good reason. However when you say fabs it knows that if it's given a integral it should convert it to floating because it only has overloads for floating numbers, this is a fix for compilation but a better fix would be to have nextChars and prevChars be signed if the use of abs was really intended by @endless-sky 

I think :>

**Comment by endless-sky** [comment]
Fixed in d91d237 and a6759a2.

**Comment by Pointedstick** [comment]
That fixed it!
